sympy/solvers/tests/test_solveset.py

```diff
@@ -177,6 +177,9 @@ def test_domain_check():
     assert domain_check(x, x, oo) is False
     assert domain_check(0, x, oo) is False
 
+def test_issue_11536():
+    assert solveset(0**x - 100, x, S.Reals) == ConditionSet(x, Eq(0**x - 100, 0), S.Reals)
+    assert solveset(0**x - 1, x, S.Reals) == FiniteSet(0)
```

Comment:
Do these cover all cases?
1. all solutions pass
1. any solution fails in checksol
1. any solution raises an exception with checksol
1. f_x does not become x
1. the solution is not a FiniteSet